fabric-enchantments-v1/src/main/java/net/fabricmc/fabric/api/enchantment/v1/FabricEnchantment.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.enchantment.v1;
+
+import net.minecraft.enchantment.EnchantmentTarget;
+
+/**
+ * Implement this interface on your enchantments if you want them to use a custom enchantment target.
+ * Your enchantments should have a vanilla enchantment target of {@link EnchantmentTarget#ALL}.
+ */
+public interface FabricEnchantment {
```

Comment:
Why not make this a class that already extends Enchanement?

This could then allow you to remove `FabricEnchantmentTarget` and add `isAcceptableItem` in here.

Reply:
That's a good question.
The only downside would be `EnchantmentTarget.ALL.isAcceptableItem` (which should check if any enchantments are applicable at all to the item). It would also get rid of the vanilla structuring.

I'm not sure about that I'm gonna explore it a bit maybe on a different branch.

Reply:
Just saw that vanilla already has `Enchantment.isAcceptableItem(ItemStack)` it's just not used for enchanting (but everywhere else lol)

Reply:
Ok, there's actually a difference between `Enchantment.isAcceptableItem` and `EnchantmentTarget.isAcceptableItem`. The first one defines that the enchantment is generally allowed on a stack. The latter is used for determining if enchantments can directly go on items when enchanting them.

Example: An axe can't be enchanted with sharpness but it can still be applied in an anvil.

Reply:
I got a vote for you:
👍 make `FabricEnchantment` a class and keep custom target functionality
😄 completely remove target stuff and move everything to `FabricEnchantment`

Reply:
The vote decided and the transition is done 😄